src/audio_transcription/transcription_nodes.py

```
@@ -419,6 +414,39 @@ def _transcribe_accumulated_audio(self, model_size: str, language: str, enable_v
             logger.warning(f"Non-critical transcription error, skipping chunk: {e}")
             return None
     
+    def _resample_to_16khz(self, audio_data: np.ndarray, original_sample_rate: int) -> np.ndarray:
+        """
+        Resample audio to 16kHz for optimal Whisper performance.
+        
+        Args:
+            audio_data: Input audio data as numpy array
+            original_sample_rate: Original sample rate of the audio
+            
+        Returns:
+            Resampled audio data at 16kHz
+        """
+        target_sample_rate = 16000
+        
+        if original_sample_rate == target_sample_rate:
+            return audio_data
+        
+        if audio_data.size == 0:
+            return audio_data
+        
+        try:
+            # Use scipy.signal.resample for high-quality resampling
+            num_samples = int(len(audio_data) * target_sample_rate / original_sample_rate)
+            resampled_audio = signal.resample(audio_data, num_samples)
+            
+            logger.debug(f"Resampled audio from {original_sample_rate}Hz to {target_sample_rate}Hz "
+                        f"({len(audio_data)} -> {len(resampled_audio)} samples)")
+            
+            return resampled_audio.astype(np.float32)
+            
+        except Exception as e:
+            logger.warning(f"Resampling failed, using original audio: {e}")
```

I think we might want to fail early here as opposed to trying to transcribe with an improper sample rate - as we've seen, this produces incoherent output

---

It turns out the sample rate does not negatively affect it that much, (it was good enough to merge). The cause of the incoherent transcriptions appears to be related to `dtype` conversions